Zilon.Core/Zilon.Core/PersonModules/HumanSurvivalModule.cs

```diff
@@ -208,5 +289,261 @@ public override void ResetStats()
+
+            for (var i = 0; i < equipmentModule.Count(); i++)
+            {
+                var equipment = equipmentModule[i];
+                if (equipment == null)
+                {
+                    continue;
+                }
+
+                var rules = equipment.Scheme.Equip.Rules;
+
+                if (rules == null)
+                {
+                    continue;
+                }
+
+                foreach (var rule in rules)
+                {
+                    switch (rule.Type)
+                    {
+                        case EquipCommonRuleType.Health:
+                            BonusToHealth(rule.Level, rule.Direction, ref bonusList);
+                            break;
+
+                        case EquipCommonRuleType.HealthIfNoBody:
+
+                            var requirementsCompleted = true;
+
+                            for (var slotIndex = 0; slotIndex < equipmentModule.Count(); slotIndex++)
+                            {
+                                if ((equipmentModule.Slots[slotIndex].Types & EquipmentSlotTypes.Body) > 0
+                                    && equipmentModule[slotIndex] != null)
+                                {
+                                    requirementsCompleted = false;
+                                    break;
+                                }
+                            }
+
+                            if (requirementsCompleted)
+                            {
+                                BonusToHealth(rule.Level, rule.Direction, ref bonusList);
+                            }
+
+                            break;
+
+                        case EquipCommonRuleType.HungerResistance:
+                            BonusToDownPass(SurvivalStatType.Satiety, rule.Level, rule.Direction, ref bonusList);
+                            break;
+
+                        case EquipCommonRuleType.ThristResistance:
+                            BonusToDownPass(SurvivalStatType.Hydration, rule.Level, rule.Direction, ref bonusList);
+                            break;
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Помещает в список бонус на ХП.
+        /// </summary>
+        /// <param name="level"> Уровень бонуса. </param>
+        /// <param name="direction"> Направление бонуса. </param>
+        /// <param name="bonuses"> Аккумулирующий список бонусов.
+        /// Отмечен ref, чтобы показать, что изменяется внутри метода. </param>
+        private void BonusToHealth(PersonRuleLevel level, PersonRuleDirection direction,
+            ref List<SurvivalStatBonus> bonuses)
+        {
+            var survivalModule = this;
+            const SurvivalStatType hpStatType = SurvivalStatType.Health;
+            var hpStat = survivalModule.Stats.SingleOrDefault(x => x.Type == hpStatType);
+            if (hpStat != null)
+            {
+                var bonus = 0;
+                switch (level)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/manual/kreghek/Zilon_Roguelike/pullRequest?prid=5480188)